src/frontend/src/optimizer/plan_node/utils.rs

```diff
@@ -462,3 +465,124 @@ pub fn to_pb_time_travel_as_of(a: &Option<AsOf>) -> Result<Option<PbAsOf>> {
         as_of_type: Some(as_of_type),
     }))
 }
+
+pub fn scan_ranges_as_strs(order_names: Vec<String>, scan_ranges: &Vec<ScanRange>) -> Vec<String> {
+    let mut range_strs = vec![];
+
+    let explain_max_range = 20;
+    for scan_range in scan_ranges.iter().take(explain_max_range) {
+        #[expect(clippy::disallowed_methods)]
+        match scan_range {
+            ScanRange::PrefixScanRange(scan_range) => {
+                let mut range_str = scan_range
+                    .eq_conds
+                    .iter()
+                    .zip(order_names.iter())
+                    .map(|(v, name)| match v {
+                        Some(v) => format!("{} = {:?}", name, v),
+                        None => format!("{} IS NULL", name),
+                    })
+                    .collect_vec();
+                if !is_full_range(&scan_range.range) {
+                    let i = scan_range.eq_conds.len();
+                    range_str.push(range_to_string(&order_names[i], &scan_range.range))
+                }
+                range_strs.push(range_str.join(" AND "));
+            }
+            ScanRange::RowScanRange(scan_range) => {
+                let range_str = match (&scan_range.range.0, &scan_range.range.1) {
+                    (Bound::Unbounded, Bound::Unbounded) => unreachable!(),
+                    (Bound::Unbounded, ub) => ub_struct_to_string(&order_names, ub),
+                    (lb, Bound::Unbounded) => lb_struct_to_string(&order_names, lb),
+                    (lb, ub) => format!(
+                        "{} AND {}",
+                        lb_struct_to_string(&order_names, lb),
+                        ub_struct_to_string(&order_names, ub)
+                    ),
+                };
+                range_strs.push(range_str);
+            }
+        }
+    }
+    if scan_ranges.len() > explain_max_range {
+        range_strs.push("...".to_string());
+    }
+    range_strs
+}
+
+pub fn ub_struct_to_string(
```

Comment:
I think we should consider change `struct` to `row`, since we use `RowScanRange` instead of `StructScanRange`